teos/src/main.rs

```diff
@@ -218,7 +215,51 @@ async fn main() {
     };
 
     let mut poller = ChainPoller::new(&mut derefed, Network::from_str(btc_network).unwrap());
-    let last_n_blocks = get_last_n_blocks(&mut poller, tip, IRREVOCABLY_RESOLVED as usize).await;
+    let last_n_blocks = match get_last_n_blocks(&mut poller, tip, IRREVOCABLY_RESOLVED as usize)
+        .await
+    {
+        Ok(blocks) => blocks,
+        Err(_) => {
```

Comment:
If we go into this branch we would return some last_n_blocks whose tip isn't the same as the `tip` variable we are using in the initialization of the tower components (watcher, etc...).